File: 2025-08/2025-08-23/unnaturalPrime.js
function isUnnaturalPrime(n) {
  n = Math.abs(n);

  if (n == 0 || n == 1) 
    return false;
  else
    for (let i = 2; i <= n / 2; i ++) {
      if (n % i == 0)
        return false;
    }

  return true;
}

console.log(isUnnaturalPrime(1));
console.log(isUnnaturalPrime(-1));
console.log(isUnnaturalPrime(19));
console.log(isUnnaturalPrime(-23));
console.log(isUnnaturalPrime(0));
console.log(isUnnaturalPrime(97));
console.log(isUnnaturalPrime(-61));
console.log(isUnnaturalPrime(99));
console.log(isUnnaturalPrime(-44));

// --- TEST SUITE ---

const testsText = `
1. isUnnaturalPrime(1) should return false.
2. isUnnaturalPrime(-1) should return false.
3. isUnnaturalPrime(19) should return true.
4. isUnnaturalPrime(-23) should return true.
5. isUnnaturalPrime(0) should return false.
6. isUnnaturalPrime(97) should return true.
7. isUnnaturalPrime(-61) should return true.
8. isUnnaturalPrime(99) should return false.
9. isUnnaturalPrime(-44) should return false.
`;

const testsRegex = /(?<number>\d+)\.\s(?<functionCall>.+) should return (?<output>.+?)\.?$/gm;
const testData = [...testsText.matchAll(testsRegex).map(match => match.groups)];

function runTests(testData) {
  console.log("--------------------------");
  console.log("🧪Starting Verification...");
  console.log("--------------------------");

  let allPassed = true;
  let failCount = 0;
  
  testData.forEach(test => {
    const functionCallOutput = eval(test.functionCall);
    const testOutput = eval(test.output);
    let comparison;

    if (Array.isArray(testOutput))
      comparison = arraysEqual(functionCallOutput, testOutput);
    else
      comparison = functionCallOutput === testOutput;

    if (comparison) {
      console.log(`${test.number}.✅PASS - Function Call: ${test.functionCall}`);
    }
    else {
      console.log(`${test.number}.❌FAIL - Function Call: ${test.functionCall}\nExpected: ${testOutput}\nGot: ${functionCallOutput}`);
      allPassed = false;
      failCount ++;
    }
  })

  console.log("----------------------------");

  if (allPassed)
    console.log("🎉SUCCESS: All tests PASSED.");
  else
    console.log(`⚠️WARNING: ${failCount}/${testData.length} tests FAILED.`);
}

function arraysEqual(a, b) {
  if (a.length !== b.length)
    return false;

  for (const index in a) {
    if (Array.isArray(a[index])) {
      if (Array.isArray(b[index])) {
        if (!arraysEqual(a[index], b[index]))
          return false;
      }
      else 
        return false;
    }

    else if (a[index] !== b[index])
      return false;
  }

  return true;
}

runTests(testData);

File: 2025-08/2025-08-23/unnatural_prime.py
def is_unnatural_prime(n):
    n = abs(n)

    if n == 0 or n == 1:
        return False
    else:
        for i in range(2, int(n / 2)):
            if n % i == 0:
                return False

    return True

print(is_unnatural_prime(1))
print(is_unnatural_prime(-1))
print(is_unnatural_prime(19))
print(is_unnatural_prime(-23))
print(is_unnatural_prime(0))
print(is_unnatural_prime(97))
print(is_unnatural_prime(-61))
print(is_unnatural_prime(99))
print(is_unnatural_prime(-44))

# --- TEST SUITE ---

tests_text = '''
1. is_unnatural_prime(1) should return False.
2. is_unnatural_prime(-1) should return False.
3. is_unnatural_prime(19) should return True.
4. is_unnatural_prime(-23) should return True.
5. is_unnatural_prime(0) should return False.
6. is_unnatural_prime(97) should return True.
7. is_unnatural_prime(-61) should return True.
8. is_unnatural_prime(99) should return False.
9. is_unnatural_prime(-44) should return False.
'''

import re

tests_regex = re.compile(r"(?P<number>\d+)\.\s(?P<function_call>.+) should return (?P<output>.+?)\.?$", re.MULTILINE)
test_data = [item.groupdict() for item in re.finditer(tests_regex, tests_text)]

def run_tests(test_data):
    print("--------------------------")
    print("🧪 Starting Verification...")
    print("--------------------------")

    all_passed = True
    fail_count = 0

    for test in test_data:
        function_call_output = eval(test['function_call'])
        test_output = eval(test['output'])
        
        if function_call_output == test_output:
            print(f"{test['number']}.✅ PASS - Function Call:\n{test['function_call']}")
        else:
            print(f"{test['number']}.❌ FAIL - Function Call:\n{test['function_call']}\nExpected:\n{test_output}\nGot:\n{function_call_output}")
            all_passed = False
            fail_count += 1

    print("----------------------------");

    if (all_passed):
        print("🎉 SUCCESS: All tests PASSED.");
    else:
        print(f"⚠️ WARNING: {fail_count}/{len(test_data)} tests FAILED.");

run_tests(test_data)